src/components/layout/CoinRateForm/index.tsx

```diff
@@ -7,116 +7,117 @@ import { Coin } from '../../../constants/coins';
+	coin: Coin;
+	isLoading?: boolean;
+	isReadOnly?: boolean;
+	personalFlow: string;
+	shareScaler: number;
+	indexVal?: number;
 }
 
 export const CoinRateForm: FC<IProps> = ({
-  value,
-  onChange,
-  onClickStart,
-  onClickStop,
-  placeholder,
-  coin,
-  isLoading,
-  isReadOnly,
-  personalFlow,
-  shareScaler,
-  indexVal,
+	value,
+	onChange,
+	onClickStart,
+	onClickStop,
+	placeholder,
+	coin,
+	isLoading,
+	isReadOnly,
+	personalFlow,
+	shareScaler,
+	indexVal,
 }) => {
-  const { t } = useTranslation();
+	const { t } = useTranslation();
 
-  // Security Deposit is 4 hours worth of stream, so (4*60*60)/(30*24*60*60) = 1/180
-  return (
-    <div className={styles.input_container}>
-      <div className={styles.input_wrap}>
-        <TextInput
-          value={value}
-          className={styles.input}
-          onChange={onChange}
-          containerClassName={styles.container_input}
-          placeholder={placeholder}
-          right={<div className={styles.right}>{`${coin}x/mo.`}</div>}
-          type="number"
-        />
-      </div>
-      <div className={styles.buttons}>
-        <div className={styles.start_wrap}>
-          <ButtonNew
-            loaderColor="white"
-            color="primary"
-            onClick={onClickStart}
-            className={styles.start}
-            disabled={isReadOnly || isLoading || !value ||
-              (((Math.floor(((parseFloat(value) / 2592000) * 1e18)
-                / shareScaler) * shareScaler) / 1e18) * 2592000) === 0}
-            isLoading={isLoading}
-            data-tip
-            data-for={`depositTooltip-${indexVal}`}
-          >
-            {t('Start')}
-            /
-            {t('Edit')}
-          </ButtonNew>
-        </div>
-        <div className={styles.stop_wrap}>
-          {parseFloat(personalFlow) > 0 && (
-            <ButtonNew
-              loaderColor="#363B55"
-              color="secondary"
-              onClick={onClickStop}
-              className={styles.stop}
-              disabled={isReadOnly || isLoading}
-              isLoading={isLoading}
-            >
-              {t('Stop')}
-            </ButtonNew>
-          )}
-        </div>
-        <div style={{ flexBasis: '100%', height: '0' }}> </div>
+	// Security Deposit is 4 hours worth of stream, so (4*60*60)/(30*24*60*60) = 1/180
+	return (
+		<div className={styles.input_container}>
+			<div className={styles.input_wrap}>
+				<TextInput
+					value={value}
+					className={styles.input}
+					onChange={onChange}
+					containerClassName={styles.container_input}
+					placeholder={placeholder}
+					right={<div className={styles.right}>{`${coin}x/mo.`}</div>}
+					type="number"
+				/>
+			</div>
+			<div className={styles.buttons}>
+				<div className={styles.start_wrap}>
+					<ButtonNew
+						loaderColor="white"
+						color="primary"
+						onClick={onClickStart}
+						className={styles.start}
+						disabled={
+							isReadOnly ||
+							isLoading ||
```

Comment:
Can we move this logic into a function, then place the function directly in disabled, this will keep the logic and UI components a bit decoupled

Reply:
not been resolved